src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -227,6 +226,7 @@ static void on_writable(void* acp, grpc_error* error) {
     error = grpc_error_set_str(error, GRPC_ERROR_STR_TARGET_ADDRESS,
                                grpc_slice_from_copied_string(ac->addr_str));
   }
+  gpr_mu_unlock(&ac->mu);
```

Comment:
Nit: this is probably fine, but another way to do this would be to grab the copied slice from `ac->addr_str` right after `done = (--ac->refs == 0)` but before unlocking. Still safe since `ac->addr_str` data we have to protect here, and this way we don't hold the lock for some non-trivial error manipulations

Reply:
Right. I mention this alternative (kinda, your is better because slices in lieu of strdup) in the commit message. Will do.

Reply:
Actually, I mention it seems more fragile because if we ever read from `ac` after unlocking, we'd end up facing the same issue. I guess I can add a comment right before "copying" `ac->addr_str`.